src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java

```diff
@@ -122,20 +122,25 @@ public void swapNodes(int valueFirst, int valueSecond) {
      * Reverse a singly linked list from a given node till the end
      *
      */
-    Node reverseList(Node node) {
-        Node prevNode = head;
-        while (prevNode.next != node) {
-            prevNode = prevNode.next;
+    public Node reverseList(Node node) {
+        //When node is null, then "next" will cause an error as we are accesing node.next, which will give a NullPointer Exception
+        if(node==null){
+            return node;
         }
-        Node prev = null, curr = node, next;
+        Node prev = null;
+        Node curr = node;
+        Node next=curr.next;
         while (curr != null) {
```

Comment:
Could you add a check for curr.next  != null here instead of the nested if?

Reply:
without checking whether curr is null , if we check for curr.next it might cause NullPointer Exception , so I will check for    
 curr != null and put next = curr.next inside the while loop